packages/react-router/src/Match.tsx

```diff
@@ -127,6 +136,10 @@ function MatchInner({ matchId }: { matchId: string }): any {
     },
   })
 
+  const Comp = route.options.component ?? router.options.defaultComponent
```

Comment:
Could this also be memoed? Or maybe pulled into `useMemo` below?

Since these values don't change very often, there is no need to recompute this.

Reply:
done